melloddy_predictor/predictor_single.py

```diff
@@ -0,0 +1,411 @@
+import sparsechem as sc
+import scipy.io
+import numpy as np
+import types
+import pandas as pd
+import torch
+import sys
+import argparse
+from scipy.sparse import csr_matrix
+from scipy.special import expit
+from collections import OrderedDict
+from enum import Enum
+import warnings
+
+
+def csr_to_torch_coo(csr_mat: csr_matrix) -> torch.Tensor:
+    """
+    Converts from scipy sparse csr matrix to a torch sparsae_coo_tensor to be submitted to the sparsechem network
+    Sparsewchem requires 
+    
+    Args:
+        csr_matx (scipy.sparse.csr_matrix) sprse csr matrix to convert
+        
+    Returns:
+        torch.sparse_coo_tensor 
+    """
+    coo_mat = csr_mat.tocoo()
+    return torch.sparse_coo_tensor(indices = np.array([coo_mat.row,coo_mat.col]), values = coo_mat.data, size = coo_mat.shape, dtype=torch.float)
+
+
+class ScModelType(Enum):
+    classification = 0
+    regression = 1
+    hybrid = 2
+
+  
+class PredictorSingle:
```

Comment:
Can't we merge this class with the Model one ?

Reply:
Yes, I could see that this could work. I wasn't sure to begin with that that would work without major backward incompatibility:
* There would simply be different predict functions to choose from. 
* The init function could be made to work with the signature as in the current model object, but with tasks maps added as additional  optional parameters. Some introspection with respect to assess the model type that is done on the single row predictor object would be done in any case and not be harmful
* task map handling would be part of the model init function, but would have of course no effect if there were no task maps

**Some things are handled however slightly differently in both classes and need to addressed before we can attempt merging the classes**
* dropout: The single row predictor object applies it directly to the network upon initialization, in the current model object dropout is handled by the sparsechem predict function.  So the question is here if we apply dropout directly to the network upon initialization, whether doing it a second time in the sparsechem predict function has some unwanted effect. 
* The single row predictor object doesn't support catalogue heads, as mapping of catalogue tasks is quite convoluted, and therefore throws an error if it is initialized with catalogue head model.

Reply:
Thanks for these advices, I won't have time to do this merge but I'll create an issue with your remarks.

Reply:
Thanks for these advices, I won't have time to do this merge but I'll create an issue with your remarks.